google-beta/resource_compute_instance_group_manager.go

```diff
@@ -345,38 +351,45 @@ func flattenFixedOrPercent(fixedOrPercent *computeBeta.FixedOrPercent) []map[str
 func getManager(d *schema.ResourceData, meta interface{}) (*computeBeta.InstanceGroupManager, error) {
 	config := meta.(*Config)
 
-	project, err := getProject(d, config)
+	id, err := parseInstanceGroupManagerId(d.Id())
 	if err != nil {
 		return nil, err
 	}
-
-	region, err := getRegion(d, config)
-	if err != nil {
-		return nil, err
+	if id.Project == "" {
+		project, err := getProject(d, config)
+		if err != nil {
+			return nil, err
+		}
+		id.Project = project
+	}
+	if id.Zone == "" {
+		id.Zone, _ = getZone(d, config)
 	}
 
 	getInstanceGroupManager := func(zone string) (interface{}, error) {
-		return config.clientComputeBeta.InstanceGroupManagers.Get(project, zone, d.Id()).Do()
+		return config.clientComputeBeta.InstanceGroupManagers.Get(id.Project, zone, id.Name).Do()
 	}
 
 	var manager *computeBeta.InstanceGroupManager
-	var e error
-	if zone, _ := getZone(d, config); zone != "" {
-		manager, e = config.clientComputeBeta.InstanceGroupManagers.Get(project, zone, d.Id()).Do()
-
-		if e != nil {
-			return nil, handleNotFoundError(e, d, fmt.Sprintf("Instance Group Manager %q", d.Get("name").(string)))
+	if id.Zone != "" {
```

Comment:
style preference: use if id.Zone == "" ... else as positive boolean if/elses are easier to parse mentally.